crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1088 @@
+use std::f64::consts::{FRAC_PI_2, PI, TAU};
+use std::f64::{EPSILON, NEG_INFINITY};
+use std::fmt::{Debug, Write};
+use std::hash::Hash;
+
+use ecow::EcoVec;
+use typst_macros::{cast, func, scope, ty};
+use typst_syntax::{Span, Spanned};
+
+use super::color::Rgba;
+use super::*;
+use crate::diag::{bail, error, SourceResult};
+use crate::eval::{array, Array, Func, IntoValue};
+use crate::geom::{ColorSpace, Smart};
+
+/// A color gradient.
+///
+/// Typst supports:
+/// - Linear gradients through the [`gradient.linear` function]($gradient.linear)
+/// - 🚧 Radial gradient will soon™️ be available
+/// - 🚧 Conic gradient will soon™️ be available
+///
+/// See the [tracking issue](https://github.com/typst/typst/issues/2282) for more
+/// details on the progress of gradients.
+///
+/// ## Stops
+///
+/// A gradient is composed of a series of stops, each stop has a color and an offset.
+/// The offset is a [ratio]($ratio) between 0% and 100% that determines the position
+/// of the stop along the gradient. The color is the color of the gradient at that
```

Comment:
```suggestion
/// of the stop along the gradient. The stop's color is the color of the gradient at that
```

A bit awkward otherwise